arrows/vxl/register_algorithms.cxx

```diff
@@ -55,26 +54,127 @@ namespace kwiver {
-  REGISTER_TYPE( vxl::image_io );
-  REGISTER_TYPE( vxl::optimize_cameras );
-  REGISTER_TYPE( vxl::triangulate_landmarks );
-  REGISTER_TYPE( vxl::match_features_constrained );
-  REGISTER_TYPE( vxl::vidl_ffmpeg_video_input );
-
-  REGISTRATION_SUMMARY();
-  return REGISTRATION_FAILURES();
+  static auto const module_name = std::string( "algorithms.vxl" );
+  if (vpm.is_module_loaded( module_name ) )
+  {
+    return;
+  }
+
+  // add factory               implementation-name       type-to-create
+  auto fact = vpm.ADD_ALGORITHM( "vxl", kwiver::arrows::vxl::bundle_adjust );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                    "" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_MODULE_NAME, module_name )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_VERSION, "1.0" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_ORGANIZATION, "Kitware Inc." )
+    ;
+
+
+  fact = vpm.ADD_ALGORITHM( "vxl_homography_guided", kwiver::arrows::vxl::close_loops_homography_guided );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                    "" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_MODULE_NAME, module_name )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_VERSION, "1.0" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_ORGANIZATION, "Kitware Inc." )
+    ;
+
+
+  fact = vpm.ADD_ALGORITHM( "vxl_plane", kwiver::arrows::vxl::estimate_canonical_transform );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                    "" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_MODULE_NAME, module_name )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_VERSION, "1.0" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_ORGANIZATION, "Kitware Inc." )
+    ;
+
+
+  fact = vpm.ADD_ALGORITHM( "vxl", kwiver::arrows::vxl::estimate_essential_matrix );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                    "" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_MODULE_NAME, module_name )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_VERSION, "1.0" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_ORGANIZATION, "Kitware Inc." )
+    ;
+
+
+  fact = vpm.ADD_ALGORITHM( "vxl", kwiver::arrows::vxl::estimate_fundamental_matrix );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                    "" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_MODULE_NAME, module_name )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_VERSION, "1.0" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_ORGANIZATION, "Kitware Inc." )
+    ;
+
+
+  fact = vpm.ADD_ALGORITHM( "vxl", kwiver::arrows::vxl::estimate_homography );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                    "" )
```

Comment:
Use this

    Use VXL (rrel) to robustly estimate a homography from matched features